api/src/exam-environment/routes/exam-environment.ts

```diff
@@ -0,0 +1,551 @@
+/* eslint-disable jsdoc/require-returns, jsdoc/require-param */
+import { type FastifyPluginCallbackTypebox } from '@fastify/type-provider-typebox';
+import { type FastifyInstance, type FastifyReply } from 'fastify';
+import jwt from 'jsonwebtoken';
+
+import * as schemas from '../schemas';
+import { mapErr, syncMapErr, UpdateReqType } from '../../utils';
+import { JWT_SECRET } from '../../utils/env';
+import {
+  checkAttemptAgainstGeneratedExam,
+  checkPrerequisites,
+  constructUserExam,
+  generateExam,
+  userAttemptToDatabaseAttemptQuestionSets,
+  validateAttempt
+} from '../utils/exam';
+import { ERRORS } from '../utils/errors';
+
+/**
+ * Wrapper for endpoints related to the exam environment desktop app.
+ *
+ * Requires exam environment authorization token to be validated.
+ */
+export const examEnvironmentValidatedTokenRoutes: FastifyPluginCallbackTypebox =
+  (fastify, _options, done) => {
+    // TODO: Is there any reason to delete the token without generating a new one?
```

Comment:
Right. The question more so came about from the old CodeAlly token logic we had on the client where you could not **just** delete your token. You clicked "regenerate token" to invalidate the old one.

Reply:
We don't need to stick to that. If it makes more sense to be able to delete a token, then we should do it. I reckon it does, since it's equivalent to signing out and that's definitely something we have to allow people to do.

Reply:
Is this something you want to defer to a followup? If so, we could leave a TODO detailing the plan.

Reply:
No. After discussing, we realised we do NOT want to ever delete the tokens.

I need to find a place to document this, though.